extensions-core/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -267,13 +267,13 @@ private Map<String, String> getPodTemplateAnnotations(Task task) throws IOExcept
   
   private Map<String, String> getJobLabels(KubernetesTaskRunnerConfig config, Task task)
   {
-    // Namespace is required, or else getOverlordNamespace() will return null, and this will not work correctly.
     Preconditions.checkNotNull(config.getNamespace(), "When using Custom Pod Templates, druid.indexer.runner.namespace cannot be null.");
+    String overlordNamespace = config.getOverlordNamespace() == null || config.getOverlordNamespace().isEmpty() ? config.getNamespace() : config.getOverlordNamespace();
```

Comment:
ist config.getoverlordnamespaces contains kinda the same logic?
would be better to have it just once

Reply:
Would agree to changing config.getOverlordNamespaces to contain that logic. Will do the changes when free.

Reply:
Create a check to not provide `null` from `config.getOverlordNamespace`. To address the rolling upgrade issue, we cannot return `namespace` if `overlordNamespace` is null/empty. Otherwise, Druid will filter out the old tasks that are running after upgrading.